plugins/modules/keycloak_realm_key.py

```diff
@@ -244,16 +247,30 @@ def main():
         name=dict(type='str', required=True),
         force=dict(type='bool', default=False),
         parent_id=dict(type='str', required=True),
-        provider_id=dict(type='str', default='rsa', choices=['rsa']),
+        provider_id=dict(type='str', default='rsa', choices=['rsa', 'rsa-enc']),
         config=dict(
             type='dict',
             options=dict(
                 active=dict(type='bool', default=True),
                 enabled=dict(type='bool', default=True),
                 priority=dict(type='int', required=True),
-                algorithm=dict(type='str', default='RS256', choices=['RS256']),
+                algorithm=dict(
+                    type="str",
+                    default="RS256",
+                    choices=[
+                        "RS256",
+                        "RS384",
+                        "RS512",
+                        "PS256",
+                        "PS384",
+                        "PS512",
+                        "RSA1_5",
+                        "RSA-OAEP",
+                        "RSA-OAEP-256",
```

Comment:
Adding these makes sense. I only had a need for RS256 and I did not want to overreach. However, I think adding one integration test per algorithm would be useful as the list of supported algorithms might change over time. Or these might be a typo in there :smile:.

Reply:
Yea I can add some basic integration tasks to check these I guess, although some algorithms belong on certain key types only. For example, you would not want to attempt to set `RSA-OAEP` on your signing key. I was hoping that the end user of this module would understand the values they are setting so that we don't have to write too much validation code. Ideally the keycloak api will return an error in these cases so we can just pass that error on. I think maybe we should tackle this in another PR though...